### ISSUE TITLE
Release basilisk v2018.06.01

### ISSUE THREAD

**Issue by wolfbeast** [MoonchildProductions/UXP#417]
@trav90 Considering a few important issues (e.g. add-on updates) it's probably a good idea to make a fresh Basilisk release.
I've tagged the repo, as the date says intended release date 06/01.

**Comment by trav90** [comment]
I’ll have a build ready and up on FTP tonight (after 00:00 UTC).

**Comment by JustOff** [comment]
I found that `blocklists.js` was accidentally not added to the preprocessor in #368:

https://github.com/MoonchildProductions/UXP/blob/5f8de423f190bbb79a62f804151bc24824fa32d8/browser/components/preferences/jar.mn#L9

so it became unusable:

https://github.com/MoonchildProductions/UXP/blob/41d68e3f70b0aabc01cf9a49c43cf1aab0f4bdce/browser/components/preferences/blocklists.js#L9-L13

I think it would be nice to fix this before the release.

---

This type of errors can be easily detected by adding [this simple check](https://github.com/JustOff/UXP/commit/0468922b50702cbaa6f1c26c7f982eb8100915f6) to `mach_commands.py`.

I can submit it as a PR if you find it useful.

**Comment by wolfbeast** [comment]
It would have been nice if you'd have made a PR to fix the issue. But no matter, I'll do so and move the release tag.

**Comment by JustOff** [comment]
Sorry, I thought the fix is so simple, that it's more important to report a problem before it's too late.

**Comment by wolfbeast** [comment]
Your "simple check" is too simple and won't catch anything but #ifdef -- I really don't think adding in a quick but much to naive check is a good idea.

I'm still of the opinion that we really should just blanket preprocess the jar stuff, but considering the resistance I got the previous times when suggesting it, I'm just going to keep my mouth shut about that.

**Comment by wolfbeast** [comment]
@trav90 New tag and fix pushed. Please use commit 69f7f9e5f1475891ce11cc4f431692f965b0cd30

**Comment by trav90** [comment]
@JustOff thanks for catching that.

@wolfbeast noted and will do.

**Comment by JustOff** [comment]
Yes, the check is simple and naive, but it works and can be improved.

And, sorry, but I'm also certainly against global preprocessing.

**Comment by mattatobin** [comment]
The responsibility of preprocessing is with whomever last edits a file. The python preporocessor is NOT smart enough to ignore strange edge cases that would bite us in the ass and modifing mozbuild and mozpack could go horribly wrong.

It would also add more build time especially on windows where process creation is terribly expensive.

This is purely a multitude of technical reasons for the "resistance".

The bottom line is if you hold the position that one cannot be bothered to manage preprocessing then one should reevaluate if they can be bothered with changing something at all.

I don't even know why safe browsing is still in the codebase at all it can't be used..

**Comment by trav90** [comment]
@wolfbeast I can't seem to connect to the FTP server (connection time out) to upload the release tarball.

I will try to connect again later tonight.

**Comment by wolfbeast** [comment]
@trav90 I don't see a problem with the FTP server. it's up and running normally. Please try again?

**Comment by wolfbeast** [comment]
As for preprocessing, I get it. I'm just sick of this biting us in the ass time and again. I don't see a reason why entries in `jar.mn` files for .js and .jsm can't just always be preprocessed. I know it's supposed to be the responsibility of whomever adds/removes conditionals to check and adjust build files, but it's something easily forgotten by anyone since it's only something to change between 0 and 1 conditionals.
JustOff's check would help, I guess, but it needs to be a lot smarter than proposed. Either way we should either make a new issue for that, and not discuss here, or just tell me to shut up and I'll just never complain again.

**Comment by trav90** [comment]
@wolfbeast I am able to connect now and the tarball is uploading now.

**Comment by JustOff** [comment]
There seems to be [another issue](https://github.com/MoonchildProductions/UXP/issues/425) with `blocklist.js`. I'm not sure if it's blocker, but ...

**Comment by trav90** [comment]
@JustOff that appears to only be an issue when SafeBrowsing is built. SafeBrowsing is not built in Basilisk.

**Comment by wolfbeast** [comment]
Not a blocker because safebrowsing isn't enabled in release.
Builds have been published now.

**Comment by JustOff** [comment]
> Not a blocker because safebrowsing isn't enabled in release.

The problem is that this error occurs when safebrowsing is **disabled**.

**Comment by wolfbeast** [comment]
yes, I corrected myself.
